python/ray/serve/schema.py

```diff
@@ -410,6 +412,84 @@ def kubernetes_dict(self, **kwargs) -> Dict:
         return config
 
 
+@PublicAPI(stability="alpha")
+class ServeDeploySchema(BaseModel, extra=Extra.forbid):
+    host: str = Field(
+        default="0.0.0.0",
+        description=(
+            "Host for HTTP servers to listen on. Defaults to "
+            '"0.0.0.0", which exposes Serve publicly. Cannot be updated once '
+            "Serve has started running. Serve must be shut down and restarted "
+            "with the new host instead."
+        ),
+    )
+    port: int = Field(
+        default=8000,
+        description=(
+            "Port for HTTP server. Defaults to 8000. Cannot be updated once "
+            "Serve has started running. Serve must be shut down and restarted "
+            "with the new port instead."
+        ),
+    )
+    applications: List[ServeApplicationSchema] = Field(
+        default=[],
+        description=("The set of Serve applications to run on the Ray cluster."),
+    )
+
+    @staticmethod
+    def get_empty_schema_dict() -> Dict:
+        """Returns an empty deploy schema dictionary.
+
+        Schema can be used as a representation of an empty Serve deploy config.
+        """
+
+        return {"applications": []}
+
+
+@DeveloperAPI
+def prepend_app_name_to_deployment_names(
+    app_schema: ServeApplicationSchema,
+) -> ServeApplicationSchema:
+    config_dict = app_schema.dict(exclude_unset=True)
+
+    if len(app_schema.deployments) > 0 and not app_schema.name == "":
+        for idx, deployment in enumerate(config_dict["deployments"]):
+            deployment["name"] = app_schema.name + "_" + deployment["name"]
+            config_dict["deployments"][idx] = deployment
+
+    return ServeApplicationSchema.parse_obj(config_dict)
+
+
+@DeveloperAPI
+def remove_app_name_from_deployment_names(
+    app_schema: ServeApplicationSchema,
+) -> ServeApplicationSchema:
+    config_dict = app_schema.dict(exclude_unset=True)
+
+    if len(app_schema.deployments) > 0 and not app_schema.name == "":
+        for idx, deployment in enumerate(config_dict["deployments"]):
+            prefix = app_schema.name + "_"
+            # This method should not be called on any config other than one
```

Comment:
Does this comment apply to the whole function? Can you add a brief version of the comment to a docstring?